lib/server/nahnojiresolver.py

```diff
@@ -21,10 +21,10 @@ def url(url):
 		ur = m.group('url')
 		if ur.find('.flv') > -1:
 			return [ur]
-		data = util.substr(util.request(ur),'<body style','</a>')
-		pattern = 'href=\"(.+?)\"[^>]+>' 
+		data = util.substr(util.request(ur),'<source type="video/mp4"','</source>')
+		pattern = 'src=\"(.+?)\"[^>]+>' 
```

Comment:
I've no problems with BeautifulSoap on html5lib. I use the function mentioned above with multie plugins with great results. You should definitely test it out.